pandas/core/series.py

```diff
@@ -2461,6 +2461,10 @@ def asof(self, where):
 
         If there is no good value, NaN is returned.
 
+        Note that this is really just a convenient shorthand for `Series.reindex`,
+        and is equivalent to `s.dropna().reindex(where, method='ffill')` for
+        an array of dates.
```

Comment:
@maximilianr That's a good catch, thanks! Previously it was date only, so the conversion was straightforward, but as you point out we do now have to check for the possibility of a string index.